### PR TITLE
module(server) : Create NodeBackend for Http #50

### PR DESCRIPTION
**Pull request by manekinekko**
module(server) : Create NodeBackend for Http #50 (WIP)

### CODE REVIEW

**Comment by gdi2290** [comment]
here’s the part of the code where it’s parsing the url

parse url:
https://github.com/pwnall/node-xhr2/blob/master/src/001-xml_http_request.coffee#L636...L648

check if valid before sending:
https://github.com/pwnall/node-xhr2/blob/master/src/001-xml_http_request.coffee#L173...L188

that’s why we need a full url. It's to blame for not allowing relative urls

**Comment by gdi2290** [comment]
related https://github.com/pwnall/node-xhr2/issues/18

**Comment by manekinekko** [comment]
thx @gdi2290

**Comment by manekinekko** [comment]
BTW, does the [http_server](https://github.com/angular/universal/blob/master/modules/server/src/http_server.ts) module have access to the preboot [options](https://github.com/angular/universal/blob/master/modules/preboot/src/interfaces/preboot_options.ts)? May be we could use these options to pass the URL?!

**Comment by gdi2290** [comment]
app configuration should be set via (pending) `Environment` service https://github.com/angular/universal/issues/44 that way we can set the baseUrl allow with meta tags etc.

**Comment by manekinekko** [comment]
Is the baseUrl considered to be the same as the one used when starting the Node server? I mean, is it the sever IP?

**Comment by gdi2290** [comment]
node-xhr2 released a new version (v0.1.3) with baseUrl support

**Comment by gdi2290** [comment]
you can remove all the trycatch calls. Can you also squash the commits into one

File: modules/server/src/helper.ts [hunk 1]
```diff
@@ -52,3 +52,9 @@ export function selectorRegExpFactory(selector: string): RegExp {
 }
 
 
+export function relativeToAbsoluteUrl(relativeUrl: string): string {
+  var _server = require('../../../../index.js').Server;
+  var _addr = _server.address();
+  var _baseUrl = `http://${ _addr.address }:${ _addr.port }`;
```
Comment (gdi2290) [inline modules/server/src/helper.ts]: this fine for now

**Comment by gdi2290** [comment]
merged via https://github.com/angular/universal/commit/37a76589151f90b6a6264b7da2d3f4a45947f17c

**Pull request closed without merging**